File: app/src/main/java/net/coding/program/common/photopick/ImageInfoExtra.java
package net.coding.program.common.photopick;

/**
 * Created by chenchao on 15/5/6.
 */
public class ImageInfoExtra {
    private ImageInfo mImageInfo;
    private int mCount = 0;
    private String mName = "";

    public ImageInfoExtra(String name, ImageInfo mImageInfo, int count) {
        mName = name;
        this.mImageInfo = mImageInfo;
        mCount = count;
    }

    public String getPath() {
        return mImageInfo.path;
    }

    public int getCount() {
        return mCount;
    }

    public String getmName() {
        return mName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ImageInfoExtra that = (ImageInfoExtra) o;

        if (mCount != that.mCount) return false;
        return mImageInfo.equals(that.mImageInfo);

    }

    @Override
    public int hashCode() {
        int result = mImageInfo.hashCode();
        result = 31 * result + mCount;
        return result;
    }
}

File: widget-pick-photo/src/main/java/net/coding/program/pickphoto/ImageInfoExtra.kt
package net.coding.program.pickphoto

import net.coding.program.common.ImageInfo

/**
 * Created by chenchao on 15/5/6.
 */
class ImageInfoExtra(name: String, val mImageInfo: ImageInfo ?, countParam: Int) {
    var count = 0
    private var mName = ""

    val path: String
        get() = mImageInfo!!.path

    init {
        mName = name
        count = countParam
    }

    fun getmName(): String {
        return mName
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as ImageInfoExtra?

        return if (count != that!!.count) false else mImageInfo == that!!.mImageInfo

    }

    override fun hashCode(): Int {
        var result = mImageInfo!!.hashCode()
        result = 31 * result + count
        return result
    }
}
